crates/node/src/commands/run.rs

```diff
@@ -26,6 +28,11 @@ pub struct ExtendedRunCmd {
     #[clap(long, value_enum, ignore_case = true)]
     pub sealing: Option<Sealing>,
 
+    // TODO. Move this into `Sealing::Instant` when https://github.com/clap-rs/clap/issues/2621 is resolved.
+    /// Finalize sealed blocks. This flag is ignored if sealing is set to `Manual`.
+    #[clap(long, default_value = "false", default_missing_value = "true")]
+    pub finalize: bool,
```

Comment:
this argument only makes sense when sealing is set to `Instant`
https://docs.rs/clap/latest/clap/struct.Arg.html#method.required_if_eq allow you to express this relationship

Reply:
Overall this is a lot of complexity when you could just extend the `Sealing` enum
```rust
enum Sealing {
  Manual,
  Instant,
  InstantFinality,
}

Reply:
yes, makes a lot of sense. I like this approach more.